### ISSUE TITLE
DllNotFoundException in custom compilation

### ISSUE THREAD

**Issue by idotta** [shimat/opencvsharp#1345]
# Summary of your issue

When I compile the library for Debian 9 (armhf) following the Ubuntu 18.04 instructions, I still get the System.DllNotFoundException:
Unable to load shared library 'OpenCVSharpExtern'.

The lib was compiled successfully and is currently located at: /usr/local/lib/libOpenCVSharpExtern.so

Is there something else I should do?

# Environment

OpenCVSharp 4.5.3.20210821
.NET 5
Debian 9 - Armhf (32)

**Comment by idotta** [comment]
# Steps to compile opencv
```
sudo apt install build-essential cmake git pkg-config libgtk-3-dev "libcanberra-gtk*"
sudo apt install libavcodec-dev libavformat-dev libswscale-dev libv4l-dev libxvidcore-dev libx264-dev
sudo apt install libjpeg-dev libpng-dev libtiff-dev gfortran openexr libatlas-base-dev opencl-headers
sudo apt install python3-dev python3-numpy libtbb2 libtbb-dev libdc1394-22-dev

mkdir ~/opencv_build && cd ~/opencv_build
git clone https://github.com/opencv/opencv.git
git clone https://github.com/opencv/opencv_contrib.git
cd ~/opencv_build/opencv && git checkout 4.5.3
cd ~/opencv_build/opencv_contrib && git checkout 4.5.3

mkdir -p ~/opencv_build/opencv/build && cd ~/opencv_build/opencv/build

cmake -D CMAKE_BUILD_TYPE=RELEASE \
    -D CMAKE_INSTALL_PREFIX=/usr/local \
    -D INSTALL_C_EXAMPLES=OFF \
    -D INSTALL_PYTHON_EXAMPLES=OFF \
    -D OPENCV_GENERATE_PKGCONFIG=ON \
    -D ENABLE_NEON=ON \
    -D OPENCV_EXTRA_EXE_LINKER_FLAGS=-latomic \
    -D ENABLE_VFPV3=ON \
    -D BUILD_TESTS=OFF \
    -D OPENCV_ENABLE_NONFREE=ON \
    -D OPENCV_EXTRA_MODULES_PATH=~/opencv_build/opencv_contrib/modules \
    -D BUILD_EXAMPLES=OFF ..

make
sudo make install
```
# Steps to compile opencvsharp
```
git clone https://github.com/shimat/opencvsharp.git
cd opencvsharp
git fetch --all --tags --prune && git checkout 4.5.3.20210821

cd opencvsharp/src
mkdir build
cd build
cmake ..
make
sudo make install
```

**Comment by sdcb** [comment]
please use `ldd` command to provide more info.

**Comment by idotta** [comment]
Thanks for the response and sorry for my delay. Here's the result:
```
 ldd ./MyAvaloniaApp
        linux-vdso.so.1 (0xbedcd000)
        libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0xb6f61000)
        libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0xb6f4e000)
        libstdc++.so.6 => /usr/lib/arm-linux-gnueabihf/libstdc++.so.6 (0xb6e42000)
        libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0xb6dca000)
        libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0xb6da1000)
        libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0xb6cb3000)
        /lib/ld-linux-armhf.so.3 (0xb6fc2000)

ldd libOpenCvSharpExtern.so
        linux-vdso.so.1 (0xbef9d000)
        libopencv_stitching.so.4.5 => /usr/local/lib/libopencv_stitching.so.4.5 (0xb6d37000)
        libopencv_aruco.so.4.5 => /usr/local/lib/libopencv_aruco.so.4.5 (0xb6cdf000)
        libopencv_barcode.so.4.5 => /usr/local/lib/libopencv_barcode.so.4.5 (0xb6cc1000)
        libopencv_bgsegm.so.4.5 => /usr/local/lib/libopencv_bgsegm.so.4.5 (0xb6ca0000)
        libopencv_bioinspired.so.4.5 => /usr/local/lib/libopencv_bioinspired.so.4.5 (0xb6c6e000)
        libopencv_ccalib.so.4.5 => /usr/local/lib/libopencv_ccalib.so.4.5 (0xb6c2c000)
        libopencv_dnn_objdetect.so.4.5 => /usr/local/lib/libopencv_dnn_objdetect.so.4.5 (0xb6c16000)
        libopencv_dnn_superres.so.4.5 => /usr/local/lib/libopencv_dnn_superres.so.4.5 (0xb6bfd000)
        libopencv_dpm.so.4.5 => /usr/local/lib/libopencv_dpm.so.4.5 (0xb6bde000)
        libopencv_face.so.4.5 => /usr/local/lib/libopencv_face.so.4.5 (0xb6b8a000)
        libopencv_freetype.so.4.5 => /usr/local/lib/libopencv_freetype.so.4.5 (0xb6b74000)
        libopencv_fuzzy.so.4.5 => /usr/local/lib/libopencv_fuzzy.so.4.5 (0xb6b5b000)
        libopencv_hfs.so.4.5 => /usr/local/lib/libopencv_hfs.so.4.5 (0xb6b41000)
        libopencv_img_hash.so.4.5 => /usr/local/lib/libopencv_img_hash.so.4.5 (0xb6b24000)
        libopencv_intensity_transform.so.4.5 => /usr/local/lib/libopencv_intensity_transform.so.4.5 (0xb6b11000)
        libopencv_line_descriptor.so.4.5 => /usr/local/lib/libopencv_line_descriptor.so.4.5 (0xb6ae9000)
        libopencv_mcc.so.4.5 => /usr/local/lib/libopencv_mcc.so.4.5 (0xb6a9b000)
        libopencv_quality.so.4.5 => /usr/local/lib/libopencv_quality.so.4.5 (0xb6a80000)
        libopencv_rapid.so.4.5 => /usr/local/lib/libopencv_rapid.so.4.5 (0xb6a65000)
        libopencv_reg.so.4.5 => /usr/local/lib/libopencv_reg.so.4.5 (0xb6a44000)
        libopencv_rgbd.so.4.5 => /usr/local/lib/libopencv_rgbd.so.4.5 (0xb6975000)
        libopencv_saliency.so.4.5 => /usr/local/lib/libopencv_saliency.so.4.5 (0xb6951000)
        libopencv_stereo.so.4.5 => /usr/local/lib/libopencv_stereo.so.4.5 (0xb692a000)
        libopencv_structured_light.so.4.5 => /usr/local/lib/libopencv_structured_light.so.4.5 (0xb690f000)
        libopencv_superres.so.4.5 => /usr/local/lib/libopencv_superres.so.4.5 (0xb68ec000)
        libopencv_surface_matching.so.4.5 => /usr/local/lib/libopencv_surface_matching.so.4.5 (0xb68af000)
        libopencv_tracking.so.4.5 => /usr/local/lib/libopencv_tracking.so.4.5 (0xb66f0000)
        libopencv_videostab.so.4.5 => /usr/local/lib/libopencv_videostab.so.4.5 (0xb66b9000)
        libopencv_wechat_qrcode.so.4.5 => /usr/local/lib/libopencv_wechat_qrcode.so.4.5 (0xb666a000)
        libopencv_xfeatures2d.so.4.5 => /usr/local/lib/libopencv_xfeatures2d.so.4.5 (0xb65f1000)
        libopencv_xobjdetect.so.4.5 => /usr/local/lib/libopencv_xobjdetect.so.4.5 (0xb65d6000)
        libopencv_xphoto.so.4.5 => /usr/local/lib/libopencv_xphoto.so.4.5 (0xb657f000)
        libopencv_shape.so.4.5 => /usr/local/lib/libopencv_shape.so.4.5 (0xb655b000)
        libopencv_highgui.so.4.5 => /usr/local/lib/libopencv_highgui.so.4.5 (0xb6522000)
        libopencv_datasets.so.4.5 => /usr/local/lib/libopencv_datasets.so.4.5 (0xb64bc000)
        libopencv_plot.so.4.5 => /usr/local/lib/libopencv_plot.so.4.5 (0xb64a7000)
        libopencv_text.so.4.5 => /usr/local/lib/libopencv_text.so.4.5 (0xb645b000)
        libopencv_ml.so.4.5 => /usr/local/lib/libopencv_ml.so.4.5 (0xb63f8000)
        libopencv_phase_unwrapping.so.4.5 => /usr/local/lib/libopencv_phase_unwrapping.so.4.5 (0xb63e1000)
        libopencv_optflow.so.4.5 => /usr/local/lib/libopencv_optflow.so.4.5 (0xb6394000)
        libopencv_ximgproc.so.4.5 => /usr/local/lib/libopencv_ximgproc.so.4.5 (0xb61df000)
        libopencv_video.so.4.5 => /usr/local/lib/libopencv_video.so.4.5 (0xb6179000)
        libopencv_videoio.so.4.5 => /usr/local/lib/libopencv_videoio.so.4.5 (0xb6116000)
        libopencv_dnn.so.4.5 => /usr/local/lib/libopencv_dnn.so.4.5 (0xb5e61000)
        libopencv_imgcodecs.so.4.5 => /usr/local/lib/libopencv_imgcodecs.so.4.5 (0xb5c68000)
        libopencv_objdetect.so.4.5 => /usr/local/lib/libopencv_objdetect.so.4.5 (0xb5c09000)
        libopencv_calib3d.so.4.5 => /usr/local/lib/libopencv_calib3d.so.4.5 (0xb5ad7000)
        libopencv_features2d.so.4.5 => /usr/local/lib/libopencv_features2d.so.4.5 (0xb5a5a000)
        libopencv_flann.so.4.5 => /usr/local/lib/libopencv_flann.so.4.5 (0xb5a13000)
        libopencv_photo.so.4.5 => /usr/local/lib/libopencv_photo.so.4.5 (0xb599e000)
        libopencv_imgproc.so.4.5 => /usr/local/lib/libopencv_imgproc.so.4.5 (0xb5675000)
        libopencv_core.so.4.5 => /usr/local/lib/libopencv_core.so.4.5 (0xb540a000)
        libstdc++.so.6 => /usr/lib/arm-linux-gnueabihf/libstdc++.so.6 (0xb52de000)
        libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0xb5266000)
        libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0xb523d000)
        libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0xb514f000)
        libfreetype.so.6 => /usr/lib/arm-linux-gnueabihf/libfreetype.so.6 (0xb50d6000)
        libharfbuzz.so.0 => /usr/lib/arm-linux-gnueabihf/libharfbuzz.so.0 (0xb505a000)
        libgtk-3.so.0 => /usr/lib/arm-linux-gnueabihf/libgtk-3.so.0 (0xb4b4f000)
        libgdk-3.so.0 => /usr/lib/arm-linux-gnueabihf/libgdk-3.so.0 (0xb4aa3000)
        libcairo.so.2 => /usr/lib/arm-linux-gnueabihf/libcairo.so.2 (0xb49eb000)
        libgdk_pixbuf-2.0.so.0 => /usr/lib/arm-linux-gnueabihf/libgdk_pixbuf-2.0.so.0 (0xb49c1000)
        libgobject-2.0.so.0 => /usr/lib/arm-linux-gnueabihf/libgobject-2.0.so.0 (0xb4979000)
        libglib-2.0.so.0 => /lib/arm-linux-gnueabihf/libglib-2.0.so.0 (0xb489e000)
        libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0xb487a000)
        librt.so.1 => /lib/arm-linux-gnueabihf/librt.so.1 (0xb4864000)
        libdc1394.so.22 => /usr/lib/arm-linux-gnueabihf/libdc1394.so.22 (0xb47ed000)
        libavcodec.so.57 => /usr/lib/arm-linux-gnueabihf/libavcodec.so.57 (0xb3835000)
        libavformat.so.57 => /usr/lib/arm-linux-gnueabihf/libavformat.so.57 (0xb369f000)
        libavutil.so.55 => /usr/lib/arm-linux-gnueabihf/libavutil.so.55 (0xb361e000)
        libswscale.so.4 => /usr/lib/arm-linux-gnueabihf/libswscale.so.4 (0xb35bd000)
        /lib/ld-linux-armhf.so.3 (0xb6f91000)
        libjpeg.so.62 => /usr/lib/arm-linux-gnueabihf/libjpeg.so.62 (0xb3581000)
        libpng16.so.16 => /usr/lib/arm-linux-gnueabihf/libpng16.so.16 (0xb3551000)
        libtiff.so.5 => /usr/lib/arm-linux-gnueabihf/libtiff.so.5 (0xb34e9000)
        libz.so.1 => /lib/arm-linux-gnueabihf/libz.so.1 (0xb34c7000)
        libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0xb34b4000)
        libgraphite2.so.3 => /usr/lib/arm-linux-gnueabihf/libgraphite2.so.3 (0xb3489000)
        libgmodule-2.0.so.0 => /usr/lib/arm-linux-gnueabihf/libgmodule-2.0.so.0 (0xb3476000)
        libpangocairo-1.0.so.0 => /usr/lib/arm-linux-gnueabihf/libpangocairo-1.0.so.0 (0xb345d000)
        libX11.so.6 => /usr/lib/arm-linux-gnueabihf/libX11.so.6 (0xb3369000)
        libXi.so.6 => /usr/lib/arm-linux-gnueabihf/libXi.so.6 (0xb334f000)
        libXcomposite.so.1 => /usr/lib/arm-linux-gnueabihf/libXcomposite.so.1 (0xb333a000)
        libXdamage.so.1 => /usr/lib/arm-linux-gnueabihf/libXdamage.so.1 (0xb3327000)
        libXfixes.so.3 => /usr/lib/arm-linux-gnueabihf/libXfixes.so.3 (0xb3313000)
        libcairo-gobject.so.2 => /usr/lib/arm-linux-gnueabihf/libcairo-gobject.so.2 (0xb32fd000)
        libatk-1.0.so.0 => /usr/lib/arm-linux-gnueabihf/libatk-1.0.so.0 (0xb32d5000)
        libatk-bridge-2.0.so.0 => /usr/lib/arm-linux-gnueabihf/libatk-bridge-2.0.so.0 (0xb32a6000)
        libxkbcommon.so.0 => /usr/lib/arm-linux-gnueabihf/libxkbcommon.so.0 (0xb3266000)
        libwayland-cursor.so.0 => /usr/lib/arm-linux-gnueabihf/libwayland-cursor.so.0 (0xb324f000)
        libwayland-egl.so.1 => /usr/lib/arm-linux-gnueabihf/libwayland-egl.so.1 (0xb323d000)
        libwayland-client.so.0 => /usr/lib/arm-linux-gnueabihf/libwayland-client.so.0 (0xb3224000)
        libepoxy.so.0 => /usr/lib/arm-linux-gnueabihf/libepoxy.so.0 (0xb3178000)
        libpangoft2-1.0.so.0 => /usr/lib/arm-linux-gnueabihf/libpangoft2-1.0.so.0 (0xb3157000)
        libpango-1.0.so.0 => /usr/lib/arm-linux-gnueabihf/libpango-1.0.so.0 (0xb3112000)
        libfontconfig.so.1 => /usr/lib/arm-linux-gnueabihf/libfontconfig.so.1 (0xb30da000)
        libgio-2.0.so.0 => /usr/lib/arm-linux-gnueabihf/libgio-2.0.so.0 (0xb2fd1000)
        libXinerama.so.1 => /usr/lib/arm-linux-gnueabihf/libXinerama.so.1 (0xb2fbe000)
        libXrandr.so.2 => /usr/lib/arm-linux-gnueabihf/libXrandr.so.2 (0xb2fa7000)
        libXcursor.so.1 => /usr/lib/arm-linux-gnueabihf/libXcursor.so.1 (0xb2f90000)
        libXext.so.6 => /usr/lib/arm-linux-gnueabihf/libXext.so.6 (0xb2f75000)
        libpixman-1.so.0 => /usr/lib/arm-linux-gnueabihf/libpixman-1.so.0 (0xb2eec000)
        libxcb-shm.so.0 => /usr/lib/arm-linux-gnueabihf/libxcb-shm.so.0 (0xb2ed9000)
        libxcb.so.1 => /usr/lib/arm-linux-gnueabihf/libxcb.so.1 (0xb2eaf000)
        libxcb-render.so.0 => /usr/lib/arm-linux-gnueabihf/libxcb-render.so.0 (0xb2e96000)
        libXrender.so.1 => /usr/lib/arm-linux-gnueabihf/libXrender.so.1 (0xb2e7f000)
        libffi.so.6 => /usr/lib/arm-linux-gnueabihf/libffi.so.6 (0xb2e69000)
        libpcre.so.3 => /lib/arm-linux-gnueabihf/libpcre.so.3 (0xb2e0a000)
        libraw1394.so.11 => /usr/lib/arm-linux-gnueabihf/libraw1394.so.11 (0xb2df0000)
        libusb-1.0.so.0 => /lib/arm-linux-gnueabihf/libusb-1.0.so.0 (0xb2dcf000)
        libswresample.so.2 => /usr/lib/arm-linux-gnueabihf/libswresample.so.2 (0xb2dad000)
        libva.so.1 => /usr/lib/arm-linux-gnueabihf/libva.so.1 (0xb2d86000)
        libzvbi.so.0 => /usr/lib/arm-linux-gnueabihf/libzvbi.so.0 (0xb2d18000)
        libxvidcore.so.4 => /usr/lib/arm-linux-gnueabihf/libxvidcore.so.4 (0xb2c44000)
        libx265.so.95 => /usr/lib/arm-linux-gnueabihf/libx265.so.95 (0xb2a73000)
        libx264.so.148 => /usr/lib/arm-linux-gnueabihf/libx264.so.148 (0xb292f000)
        libwebpmux.so.2 => /usr/lib/arm-linux-gnueabihf/libwebpmux.so.2 (0xb2919000)
        libwebp.so.6 => /usr/lib/arm-linux-gnueabihf/libwebp.so.6 (0xb28ce000)
        libwavpack.so.1 => /usr/lib/arm-linux-gnueabihf/libwavpack.so.1 (0xb28a2000)
        libvpx.so.4 => /usr/lib/arm-linux-gnueabihf/libvpx.so.4 (0xb27db000)
        libvorbisenc.so.2 => /usr/lib/arm-linux-gnueabihf/libvorbisenc.so.2 (0xb2749000)
        libvorbis.so.0 => /usr/lib/arm-linux-gnueabihf/libvorbis.so.0 (0xb2718000)
        libtwolame.so.0 => /usr/lib/arm-linux-gnueabihf/libtwolame.so.0 (0xb26eb000)
        libtheoraenc.so.1 => /usr/lib/arm-linux-gnueabihf/libtheoraenc.so.1 (0xb26af000)
        libtheoradec.so.1 => /usr/lib/arm-linux-gnueabihf/libtheoradec.so.1 (0xb268b000)
        libspeex.so.1 => /usr/lib/arm-linux-gnueabihf/libspeex.so.1 (0xb2669000)
        libsnappy.so.1 => /usr/lib/arm-linux-gnueabihf/libsnappy.so.1 (0xb2653000)
        libshine.so.3 => /usr/lib/arm-linux-gnueabihf/libshine.so.3 (0xb263a000)
        libopus.so.0 => /usr/lib/arm-linux-gnueabihf/libopus.so.0 (0xb25f4000)
        libopenjp2.so.7 => /usr/lib/arm-linux-gnueabihf/libopenjp2.so.7 (0xb25bc000)
        libmp3lame.so.0 => /usr/lib/arm-linux-gnueabihf/libmp3lame.so.0 (0xb254d000)
        libgsm.so.1 => /usr/lib/arm-linux-gnueabihf/libgsm.so.1 (0xb2535000)
        liblzma.so.5 => /lib/arm-linux-gnueabihf/liblzma.so.5 (0xb250b000)
        libssh-gcrypt.so.4 => /usr/lib/arm-linux-gnueabihf/libssh-gcrypt.so.4 (0xb24ad000)
        libopenmpt.so.0 => /usr/lib/arm-linux-gnueabihf/libopenmpt.so.0 (0xb2384000)
        libgme.so.0 => /usr/lib/arm-linux-gnueabihf/libgme.so.0 (0xb2345000)
        libbluray.so.1 => /usr/lib/arm-linux-gnueabihf/libbluray.so.1 (0xb2303000)
        libgnutls.so.30 => /usr/lib/arm-linux-gnueabihf/libgnutls.so.30 (0xb21d2000)
        libchromaprint.so.1 => /usr/lib/arm-linux-gnueabihf/libchromaprint.so.1 (0xb21b5000)
        libbz2.so.1.0 => /lib/arm-linux-gnueabihf/libbz2.so.1.0 (0xb2199000)
        libvdpau.so.1 => /usr/lib/arm-linux-gnueabihf/libvdpau.so.1 (0xb2186000)
        libva-drm.so.1 => /usr/lib/arm-linux-gnueabihf/libva-drm.so.1 (0xb2173000)
        libva-x11.so.1 => /usr/lib/arm-linux-gnueabihf/libva-x11.so.1 (0xb215f000)
        libjbig.so.0 => /usr/lib/arm-linux-gnueabihf/libjbig.so.0 (0xb2144000)
        libgthread-2.0.so.0 => /usr/lib/arm-linux-gnueabihf/libgthread-2.0.so.0 (0xb2132000)
        libatspi.so.0 => /usr/lib/arm-linux-gnueabihf/libatspi.so.0 (0xb2101000)
        libdbus-1.so.3 => /lib/arm-linux-gnueabihf/libdbus-1.so.3 (0xb20c0000)
        libthai.so.0 => /usr/lib/arm-linux-gnueabihf/libthai.so.0 (0xb20a9000)
        libexpat.so.1 => /lib/arm-linux-gnueabihf/libexpat.so.1 (0xb2081000)
        libselinux.so.1 => /lib/arm-linux-gnueabihf/libselinux.so.1 (0xb2057000)
        libresolv.so.2 => /lib/arm-linux-gnueabihf/libresolv.so.2 (0xb2037000)
        libmount.so.1 => /lib/arm-linux-gnueabihf/libmount.so.1 (0xb1ff5000)
        libXau.so.6 => /usr/lib/arm-linux-gnueabihf/libXau.so.6 (0xb1fea000)
        libXdmcp.so.6 => /usr/lib/arm-linux-gnueabihf/libXdmcp.so.6 (0xb1fd6000)
        libudev.so.1 => /lib/arm-linux-gnueabihf/libudev.so.1 (0xb1fbd000)
        libsoxr.so.0 => /usr/lib/arm-linux-gnueabihf/libsoxr.so.0 (0xb1f60000)
        libogg.so.0 => /usr/lib/arm-linux-gnueabihf/libogg.so.0 (0xb1f53000)
        libgcrypt.so.20 => /lib/arm-linux-gnueabihf/libgcrypt.so.20 (0xb1ea8000)
        libgssapi_krb5.so.2 => /usr/lib/arm-linux-gnueabihf/libgssapi_krb5.so.2 (0xb1e6e000)
        libmpg123.so.0 => /usr/lib/arm-linux-gnueabihf/libmpg123.so.0 (0xb1e1b000)
        libvorbisfile.so.3 => /usr/lib/arm-linux-gnueabihf/libvorbisfile.so.3 (0xb1e04000)
        libxml2.so.2 => /usr/lib/arm-linux-gnueabihf/libxml2.so.2 (0xb1ce6000)
        libp11-kit.so.0 => /usr/lib/arm-linux-gnueabihf/libp11-kit.so.0 (0xb1c9a000)
        libidn.so.11 => /lib/arm-linux-gnueabihf/libidn.so.11 (0xb1c58000)
        libtasn1.so.6 => /usr/lib/arm-linux-gnueabihf/libtasn1.so.6 (0xb1c3c000)
        libnettle.so.6 => /usr/lib/arm-linux-gnueabihf/libnettle.so.6 (0xb1bfd000)
        libhogweed.so.4 => /usr/lib/arm-linux-gnueabihf/libhogweed.so.4 (0xb1bc5000)
        libgmp.so.10 => /usr/lib/arm-linux-gnueabihf/libgmp.so.10 (0xb1b6c000)
        libdrm.so.2 => /usr/lib/arm-linux-gnueabihf/libdrm.so.2 (0xb1b51000)
        libsystemd.so.0 => /lib/arm-linux-gnueabihf/libsystemd.so.0 (0xb1af7000)
        libdatrie.so.1 => /usr/lib/arm-linux-gnueabihf/libdatrie.so.1 (0xb1ae3000)
        libblkid.so.1 => /lib/arm-linux-gnueabihf/libblkid.so.1 (0xb1aa7000)
        libbsd.so.0 => /lib/arm-linux-gnueabihf/libbsd.so.0 (0xb1a84000)
        libgomp.so.1 => /usr/lib/arm-linux-gnueabihf/libgomp.so.1 (0xb1a54000)
        libgpg-error.so.0 => /lib/arm-linux-gnueabihf/libgpg-error.so.0 (0xb1a37000)
        libkrb5.so.3 => /usr/lib/arm-linux-gnueabihf/libkrb5.so.3 (0xb199f000)
        libk5crypto.so.3 => /usr/lib/arm-linux-gnueabihf/libk5crypto.so.3 (0xb1969000)
        libcom_err.so.2 => /lib/arm-linux-gnueabihf/libcom_err.so.2 (0xb1956000)
        libkrb5support.so.0 => /usr/lib/arm-linux-gnueabihf/libkrb5support.so.0 (0xb193f000)
        libkeyutils.so.1 => /lib/arm-linux-gnueabihf/libkeyutils.so.1 (0xb192c000)
        libicui18n.so.57 => /usr/lib/arm-linux-gnueabihf/libicui18n.so.57 (0xb177f000)
        libicuuc.so.57 => /usr/lib/arm-linux-gnueabihf/libicuuc.so.57 (0xb1656000)
        libicudata.so.57 => /usr/lib/arm-linux-gnueabihf/libicudata.so.57 (0xafdc9000)
        liblz4.so.1 => /usr/lib/arm-linux-gnueabihf/liblz4.so.1 (0xafdab000)
        libuuid.so.1 => /lib/arm-linux-gnueabihf/libuuid.so.1 (0xafd97000)
```

**Comment by idotta** [comment]
I was just able to run my application successfully, after I followed these steps:

- Recompile opencv and opencvsharp
- Publish application (which has nuget packages for runtimes linux-arm and win) for Linux-Arm
- Exchange libOpenCvSharpExtern.so added by nuget with the one I built

I'm not sure what step actually fixed the problem, but these are easily done if anyone else encounters the same issue.

I'll close this issue then. Thanks for the support!
